common/src/main/resources/assets/ioticblocks/lang/zh_cn.flatten.json5

```diff
@@ -0,0 +1,58 @@
+{
+  ioticblocks: {
+    category: {
+      patterns: {
+        entry: {
+          readwrite_blocks: {
+            "": "读取/写入方块",
+            page: {
+              "read/block": "与$(l:patterns/readwrite#hexcasting:read)$(action)书吏之精思/$类似，但会从方块中读出 iota，而非从我另一只手中的物品读取。",
+              "write/block": "与$(l:patterns/readwrite#hexcasting:write)$(action)书吏之策略/$类似，但会向方块写入 iota，而非向我另一只手中的物品写入。$(br2)\
+                有趣的是，我似乎无法用这个法术写入我的真名。我有种感觉，如果我真可以这么做，等着我的只会是灾难。",
+              "read/index": "从实体或方块中读出一个列表 iota，并将 iota 压入栈中给定下标处。$(br2)\
+                这个图案和后页的图案似乎很适合处理大批量的数据，而且不会因此招致 $(l:casting/mishaps)Iota 过大/$事故。",
+              "write/index": "从实体或方块中读出一个列表 iota，并将该列表中给定下标处的元素修改为所给 iota，最后把修改后的列表写入原实体或方块。",
+              "readable/block": "与$(l:patterns/readwrite#hexcasting:readable)$(action)审计员之精思/$类似，但会检验方块的是否可读取，而非检验我另一只手中的物品。",
+              "writable/block": "与$(l:patterns/readwrite#hexcasting:writable)$(action)估价员之精思/$类似，但会检验方块的是否可写入，而非检验我另一只手中的物品。",
+            }
+          }
+        }
+      }
+    }
+  },
+
+  hexcasting: {
+    action: {
+      "ioticblocks:": {
+        "read/index": "速记员之馏化",
+        "write/index": "速记员之策略",
+      },
+
+      book: {
+        "ioticblocks:": {
+          "read/index": "速记员之馏化",
+        },
+      },
```

Comment:
note: I think technically this section could be removed. `hexcasting.action.book.{id}` is used to override the value of `hexcasting.action.{id}` in the ingame book if the latter is too long to fit, so I think it would be fine to remove `hexcasting.action.book.ioticblocks:read/index` here since the translation is identical in zh_cn. up to you though, it's functionally equivalent either way

Reply:
I believe it might be better to leave it to keep file structures.
It's probably a "habit" carried over from other Patchouli books I've translated. They use such keys in a "hardcoded" way from `assets`' perspective, (or if those keys weren't translated, so wouldn't they in the book), it's a fallback mechanic to keep keys present.